src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -213,13 +215,14 @@ public void OnTimeout(TimeoutReason reason)
             // TODO what timeouts should we handle here? Is keep alive something we should care about?
             switch (reason)
             {
-                case TimeoutReason.KeepAlive:
-                    SendGoAway(_highestOpenedStreamId).Preserve();
-                    break;
                 case TimeoutReason.TimeoutFeature:
                     SendGoAway(_highestOpenedStreamId).Preserve();
                     break;
                 case TimeoutReason.RequestHeaders:
+                    Log.ConnectionBadRequest(ConnectionId, KestrelBadHttpRequestException.GetException(RequestRejectionReason.RequestHeadersTimeout));
+                    Abort(new ConnectionAbortedException(CoreStrings.BadRequest_RequestHeadersTimeout), Http3ErrorCode.RequestRejected);
```

Comment:
General question for exceeding limits and timeouts in HTTP/3:

Is it a connection error when a stream exceeds a timeout or limit, or a stream error? @halter73 @Tratcher @jkotalik

Reply:
Some things like rate are calculated in aggregate. The request header timeout is specific to one request. It is simple enough to kill that one particular stream rather than the entire connection.